### PR TITLE
Fixing AutoFlusher logger string and making it blocking

### PR DESCRIPTION
**Pull request by rob-c**
#320 and have made auto-flusher blocking to see if it causes lockups. There is always the option to revert if problems arise.

### CODE REVIEW

File: python/Ganga/Core/GangaRepository/Registry.py [hunk 1]
```diff
@@ -180,6 +180,7 @@ def __init__(self, registry, *args, **kwargs):
         super(RegistryFlusher, self).__init__(*args, **kwargs)
         self.registry = registry
         self._stop = threading.Event()
+        self.logger = getLogger()
```
Comment (milliams) [inline python/Ganga/Core/GangaRepository/Registry.py]: Is there not already a global logger?

Comment (rob-c) [inline python/Ganga/Core/GangaRepository/Registry.py]: There is, I just wanted to have a separate one for the flusher. Can revert this if there's a strong reason to use the same one as the Registry.

Comment (milliams) [inline python/Ganga/Core/GangaRepository/Registry.py]: They will be indistinguishable from each other as they will have the same name I believe and in fact might even just be references to the same object.

**Comment by rob-c** [comment]
Any objections to merging this in it's current state? I've incorporated changes from #320 into this branch and all of the tests are passing.

**Comment by milliams** [comment]
Assuming that you're still not seeing any LHCb lock ups then I'm happy for this to be merged.

**Comment by rob-c** [comment]
I'm not seeing any from 'normal' use in testing so I'm keen to get this fix in to stop the debug logger getting spammed with stacktraces if nothing else.

**Pull request merged**